rali/rali/source/meta_node_ssd_random_crop.cpp

```diff
@@ -0,0 +1,237 @@
+    {
+        _batch_size = input_meta_data->size();
+    }
+    set_threshold(_node->get_threshold());
+    set_num_of_attempts(_node->get_num_of_attempts());
+    _meta_crop_param = _node->get_crop_param();
+    _dst_width = _node->get_dst_width();
+    _dst_height = _node->get_dst_height();
+    _crop_width = _meta_crop_param->cropw_arr;
+    _crop_height = _meta_crop_param->croph_arr;
+    _x1 = _meta_crop_param->x1_arr;
+    _y1 = _meta_crop_param->y1_arr;
+    _x2 = _meta_crop_param->x2_arr;
+    _y2 = _meta_crop_param->y2_arr;
+    vxCopyArrayRange((vx_array)_crop_width, 0, _batch_size, sizeof(uint), _crop_width_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_crop_height, 0, _batch_size, sizeof(uint), _crop_height_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_x1, 0, _batch_size, sizeof(uint), _x1_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_y1, 0, _batch_size, sizeof(uint), _y1_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_x2, 0, _batch_size, sizeof(uint), _x2_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_y2, 0, _batch_size, sizeof(uint), _y2_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+
+    x_drift_factor = _meta_crop_param->get_x_drift_factor();
+    y_drift_factor = _meta_crop_param->get_y_drift_factor();
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dis(0, 6);
+    const std::vector<std::pair<float, float>> IOU = {std::make_pair(0.0f, 1.0f), std::make_pair(0.1f, 1.0f), std::make_pair(0.3f, 1.0f),
+                                            std::make_pair(0.5f, 1.0f), std::make_pair(0.45f, 1.0f), std::make_pair(0.35f, 1.0f), std::make_pair(0.0f, 1.0f) };   
+    int sample_option;
+    std::pair<float, float> iou;
+    float min_iou, max_iou;
+    in_width = _meta_crop_param->in_width;
+    in_height = _meta_crop_param->in_height;
+    bool invalid_bboxes = true;
+    _enitire_iou = true;
+    BoundingBoxCord crop_box, jth_box;
+    for (int i = 0; i < _batch_size; i++)
+    {
+        int bb_count = input_meta_data->get_bb_labels_batch()[i].size();
+        std::vector<int> labels_buf(bb_count);
+        memcpy(labels_buf.data(), input_meta_data->get_bb_labels_batch()[i].data(), sizeof(int) * bb_count);
+        std::vector<float> coords_buf(bb_count * 4);
+        memcpy(coords_buf.data(), input_meta_data->get_bb_cords_batch()[i].data(), input_meta_data->get_bb_cords_batch()[i].size() * sizeof(BoundingBoxCord));
+        crop_box.h = _crop_height_val[i];
+        crop_box.w = _crop_width_val[i];
+        crop_box.x = _x1_val[i];
+        crop_box.y = _y1_val[i];
+        while (true)
+        {
+            sample_option = dis(gen);
+            iou =  IOU[sample_option];
+            min_iou = iou.first;
+            max_iou = iou.second;
+            sample_option = 0;
+            if (!sample_option)
```

Comment:
Codacy found an issue: [Condition '!sample_option' is always true](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5976200)